src/Workspaces/Core/Portable/LanguageServices/SemanticsFactsService/ISemanticFactsService.cs

```diff
@@ -99,5 +102,32 @@ internal interface ISemanticFactsService : ILanguageService
         bool IsPartial(ITypeSymbol typeSymbol, CancellationToken cancellationToken);
 
         IEnumerable<ISymbol> GetDeclaredSymbols(SemanticModel semanticModel, SyntaxNode memberDeclaration, CancellationToken cancellationToken);
+
+        SyntaxToken GenerateUniqueName(
+            SemanticModel semanticModel, SyntaxNode location, 
+            SyntaxNode containerOpt, string baseName, CancellationToken cancellationToken);
+    }
+
+    internal abstract class AbstractSemanticFactsService
+    {
+        protected abstract ISyntaxFactsService SyntaxFactsService { get; }
+
+        public SyntaxToken GenerateUniqueName(
```

Comment:
would a better place for this be inside NameGenerator?

Reply:
NameGenerator is designed to be Roslyn agnostic.  i.e. it's literally jsut string manipulation.  we've deliberately kept it that way so that partners can use it/copy it trivially.